doc/source/iocs.rst

```diff
@@ -177,6 +177,31 @@ update rate controlled by a second PV, ``random_walk:dt``.
 
 .. literalinclude:: ../../caproto/ioc_examples/random_walk.py
 
+.. note::
+
+   **What is async_lib.library.sleep?**
+
+   As caproto supports three different async libraries, we have an "async
+   layer" class that gives compatible async versions of commonly used
+   synchronization primitives. The attribute ``async_lib.library`` would be either
+   the Python module ``asyncio`` (default), ``trio``, or ``curio``, depending on how
+   the server is run.  It happens that all three of these modules have a ``sleep``
+   function at top level, so ``async_lib.library.sleep`` accesses the appropriate
```

Comment:
I think you are right about adding the hyphen to "commonly-used synchronization primitives" above because comonly-used is an adjective phrase ("used" describes primitives) but here I think we leave the hyphen out because level is a noun:

function at the top level

whereas these constructions would get a hyphen:

function at the top-level scope
top-level function

Edited locally to add missing "the". :+1:

Reply:
🤷‍♂ 👍